pkgs/top-level/aliases.nix

```diff
@@ -139,6 +139,7 @@ mapAliases ({
   draftsight = throw "draftsight has been removed, no longer available as freeware"; # added 2020-08-14
   dvb_apps = throw "dvb_apps has been removed."; # added 2020-11-03
   dwarf_fortress = dwarf-fortress; # added 2016-01-23
+  ec2instanceconnectcli = python3Packages.ec2instanceconnectcli; # added 2021-01-23
```

Comment:
I am not sure if this is the right approach. I am not sure what to add to all-packages.nix. @jonringer can you help out?

Reply:
As I understand it, in situations such as this, `aliases.nix` is preferred over `all-packages.nix`. I'm curious to see what Jon or someone else says.

Reply:
This is not really an alias per see. Everything written in aliases.nix is marked for deprecation and can't be used in other packages inputs.

Reply:
My thinking is that `ec2instanceconnectcli` is an alias for `nodePackages.ec2instanceconnectcli`. Is that incorrect?

Reply:
Yes but if some script in a derivation wants to use it, it could not do that with ofborg or hydra evaluating it. If ``ec2instanceconnectcli`` should be a real package which can be used in other things this needs to change. aliases.nix is only meant to notify end users of breakages in their config or silently redirect them to small name changes.

Reply:
when exporting an application from a python package set, you should do
```
  ec2instanceconnectcli = with python3Packages; toPythonApplication ec2instanceconnectcli;
```

otherwise the package will "taint" a user's environment with PYTHONPATH and might be incompatible with a different python interpreter.

Reply:
alias.nix is usually used to deprecated something, and it's conditionally disabled with `config.allowAliases = false;`

Reply:
Thanks for the feedback. I'll update accordingly.